renderdoc/driver/gl/gl_hooks_linux.cpp

```diff
@@ -197,6 +195,160 @@ class OpenGLHook : LibraryHook, public GLPlatform
       glXDestroyContext_real(context.dpy, context.ctx);
   }
 
+  void DeleteReplayContext(GLWindowingData context)
+  {
+    if(glXDestroyContext_real)
+    {
+      glXMakeContextCurrent_real(context.dpy, 0L, 0L, NULL);
+      glXDestroyContext_real(context.dpy, context.ctx);
+    }
+  }
+
+  void SwapBuffers(GLWindowingData context) { glXSwapBuffers_real(context.dpy, context.wnd); }
+  void GetOutputWindowDimensions(GLWindowingData context, int32_t &w, int32_t &h)
+  {
+    PFNGLXQUERYDRAWABLEPROC glXQueryDrawableProc =
+        (PFNGLXQUERYDRAWABLEPROC)dlsym(RTLD_NEXT, "glXQueryDrawable");
```

Comment:
This function will be called every time we render to a window, I think it's maybe better to maybe make this variable static we can cache the dlsym result. It shouldn't change over the lifetime of a program because it's not a `glXGetProcAddress`'d function.